parsl/executors/status_handling.py

```diff
@@ -234,3 +235,18 @@ def _get_block_and_job_ids(self) -> Tuple[List[str], List[Any]]:
     @abstractproperty
     def workers_per_node(self) -> Union[int, float]:
         pass
+
+    def create_monitoring_info(self, status: Dict[str, JobStatus]) -> Sequence[object]:
+        """Create a monitoring message for each block based on the poll status.
+        """
+        msg = []
+        for bid, s in status.items():
+            d: Dict[str, Any] = {}
+            d['run_id'] = self.run_id
+            d['status'] = s.status_name
+            d['timestamp'] = datetime.datetime.now()
+            d['executor_label'] = self.label
+            d['job_id'] = self.blocks_to_job_id.get(bid, None)
+            d['block_id'] = bid
+            msg.append(d)
+        return msg
```

Comment:
I recognize this is largely copying, but could this be reduced to a single list comprehension?  Consider:
```python
# are all items effectively collected at the same time?  Or is the loop
# iteration time pertinent, perhaps for concurrent data structure modification?
now = datetime.datetime.now()
return [
    {
        'run_id': self.run_id,
        ...
        'timestamp': now,
        ...
    }
    for bid, s in status.items()
]
```

Reply:
sure. eventually I'd also like these monitoring messages to be typed more strongly: they have strongly defined formats that are not declared in the type system but sort of are implied by the SQL schema declared in SQLAlchemy deep in the monitoring DB manager...